frappe/website/doctype/personal_data_deletion_request/personal_data_deletion_request.py

```diff
@@ -321,6 +336,25 @@ def __redact_partial_match_data(self, doctype):
 			{"name": self.full_name, "email": self.email},
 		)
 
+	@frappe.whitelist()
+	def put_on_hold(self):
+		self.db_set("status", "On Hold")
+
+def auto_delete():
+	auto_account_deletion = frappe.db.get_single_value("Website Settings", "auto_account_deletion")
+	if auto_account_deletion < 1:
+		return
+
+	requests = frappe.get_all("Personal Data Deletion Request",
+		filters = {
+			"status": "Pending Approval"
+		},
+		pluck="name")
+
+	for request in requests:
+		doc = frappe.get_doc("Personal Data Deletion Request", request)
+		if date_diff(get_datetime(), doc.creation) >= auto_account_deletion:
+			doc.trigger_data_deletion()
```

Comment:
Let's also add a comment that the user record for this request has been auto-deleted due to inactivity?